packages/deployment-server-ui/src/pages/TypeId.tsx

```diff
@@ -1,68 +1,119 @@
-import { DeployRequest, DeviceType, Device } from "deployment-server";
+import { DeployRequest, Device, DeviceTypeResource } from "deployment-server";
 import React, { useEffect, useState } from "react";
 import defineFunctionalComponent from "../util/defineFunctionalComponent";
-import { Card, Col, Row } from "antd";
+import { Card, Col, Divider, Row, Spin, Statistic } from "antd";
 import { useInterval } from "react-use";
 import { useParams } from "react-router-dom";
 import { StatusTag } from "../components/StatusTag";
+import DeploymentsByTypeGraph from "../components/Graphs/DeploymentsByTypeGraph";
+import DeploymentsStatusGraph from "../components/Graphs/DeploymentsStatusGraph";
+import IssueCountGraph from "../components/Graphs/IssueCountGraph";
+import Meta from "antd/lib/card/Meta";
 
 export default defineFunctionalComponent(function TypeId() {
-  let [devices, setDevices] = useState<Device[]>([]);
-  let [deviceType, setDeviceType] = useState<DeviceType | null>(null);
-  let [deployments, setDeployments] = useState<DeployRequest[]>([]);
-  let [refetchFlip, setRefetchFlip] = useState(false);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [devices, setDevices] = useState<Device[]>([]);
+  const [deviceType, setDeviceType] = useState<DeviceTypeResource>();
+  const [deployments, setDeployments] = useState<DeployRequest[]>([]);
+  const [refetchFlip, setRefetchFlip] = useState(false);
 
-  const { id } = useParams()
+  const { id } = useParams();
   function findDeviceById(id: string) {
-    return devices.find((device) => id === device.id)
+    return devices.find((device) => id === device.id);
   }
+
   useInterval(function () {
     setRefetchFlip(!refetchFlip);
-  }, 5000);
+  }, 1000);
 
   useEffect(() => {
-    fetch(`/api/device-types/${id}`).then(async (res) => {
-      setDeviceType(await res.json());
-    });
+    const fetchAsync = async () => {
+      const typesRes = await fetch(`/api/device-types/${id}`);
+      const types = await typesRes.json();
+      setDeviceType(types);
+
+      const devRes = await fetch(`/api/devices`);
+      const devs = await devRes.json();
+      setDevices(devs.filter((dev: Device) => dev.deviceTypeId === id));
+
+      const deployRes = await fetch("/api/deployments");
+      const deploys = await deployRes.json();
+      setDeployments(
+        deploys.filter(
+          // @ts-ignore: Unreachable code error
```

Comment:
Fix

Reply:
suggestion?

Reply:
I suspect you'll have to type the `deploys` variable?